File: Bindings~/java/src/main/java/ro/altom/alttester/Commands/InputActions/AltTilt.java
package ro.altom.alttester.Commands.InputActions;

import ro.altom.alttester.IMessageHandler;
import ro.altom.alttester.Commands.AltBaseCommand;

/**
 * Simulates device rotation action in your game.
 */
public class AltTilt extends AltBaseCommand {
    private AltTiltParams altTiltParameters;

    public AltTilt(IMessageHandler messageHandler, AltTiltParams altTiltParameters) {
        super(messageHandler);
        this.altTiltParameters = altTiltParameters;
    }

    public void Execute() {
        SendCommand(altTiltParameters);
        String data = recvall(altTiltParameters, String.class);
        validateResponse("Ok", data);

        if (altTiltParameters.getWait()) {
            data = recvall(altTiltParameters, String.class);
            validateResponse("Finished", data);
        }
    }
}

File: Assets/AltTester/Runtime/AltDriver/Commands/InputActions/AltTilt.cs
/*
    Copyright(C) 2026 Altom Consulting
*/

namespace AltTester.AltTesterSDK.Driver.Commands
{
    public class AltTilt : AltBaseCommand
    {
        AltTiltParams cmdParams;
        public AltTilt(IDriverCommunication commHandler, AltVector3 acceleration, float duration, bool wait) : base(commHandler)
        {
            cmdParams = new AltTiltParams(acceleration, duration, wait);
        }
        public void Execute()
        {
            CommHandler.Send(cmdParams);
            string data = CommHandler.Recvall<string>(cmdParams);
            ValidateResponse("Ok", data);
            if (cmdParams.wait)
            {
                data = CommHandler.Recvall<string>(cmdParams);
                ValidateResponse("Finished", data);
            }
        }
    }
}
